contracts/rewards/AutoCompoundingRewards.sol

```diff
@@ -281,26 +304,80 @@ contract AutoCompoundingRewards is IAutoCompoundingRewards, ReentrancyGuardUpgra
         emit ProgramEnabled({ pool: pool, status: status, remainingRewards: p.remainingRewards });
     }
 
+    /**
+     * @inheritdoc IAutoCompoundingRewards
+     */
+    function setAutoProcessRewardsCount(uint256 newAutoProcessRewardsCount)
+        external
+        greaterThanZero(newAutoProcessRewardsCount)
+        onlyAdmin
+    {
+        uint256 prevAutoProcessRewardsCount = _autoProcessRewardsCount;
+        if (prevAutoProcessRewardsCount == newAutoProcessRewardsCount) {
+            return;
+        }
+
+        _autoProcessRewardsCount = newAutoProcessRewardsCount;
+
+        emit AutoProcessRewardsCountUpdated({
+            prevAutoProcessRewardsCount: prevAutoProcessRewardsCount,
+            newAutoProcessRewardsCount: newAutoProcessRewardsCount
+        });
+    }
+
+    /**
+     * @inheritdoc IAutoCompoundingRewards
+     */
+    function autoProcessRewards() external nonReentrant {
+        uint256 numOfPools = _pools.length();
+        uint256 index = _nextProcessRewardsIndex;
+        uint256 count = _autoProcessRewardsCount;
+        uint256 maxCount = Math.min(count * 2, numOfPools);
+
+        for (uint256 i = 0; i < maxCount; i++) {
+            index = (index + i) % numOfPools;
+            if (_processRewards(Token(_pools.at(index)), true)) {
+                count -= 1;
+                if (count == 0) {
+                    break;
+                }
+            }
+        }
+
+        _nextProcessRewardsIndex = index + 1;
+    }
+
     /**
      * @inheritdoc IAutoCompoundingRewards
      */
     function processRewards(Token pool) external nonReentrant {
+        _processRewards(pool, false);
+    }
+
+    /**
+     * @dev processes the rewards of a given pool
+     */
+    function _processRewards(Token pool, bool skipRecent) private returns (bool) {
         ProgramData memory p = _programs[pool];
 
         uint32 currTime = _time();
 
         if (!p.isEnabled || currTime < p.startTime) {
-            return;
+            return false;
+        }
+
+        if (skipRecent && currTime < p.prevDistributionTimestamp + AUTO_PROCESS_REWARDS_MIN_TIME_DELTA) {
+            return false;
         }
 
         uint256 tokenAmountToDistribute = _tokenAmountToDistribute(p, currTime);
         if (tokenAmountToDistribute == 0) {
-            return;
+            return false;
```

Comment:
IMHO it should return `true` here, since it does indeed like a program was processed (just didn't yield any rewards; i.e., it wasn't actually skipped)

Reply:
It a product decision, I suppose.
For sure, in terms of user-perspective-efficiency, it means fewer actually-processed programs.
@yudilevi : WDYT?

Reply:
yup, should return true here (wasn't skipped but rather just didn't have what to distribute - but was fully processed)

Reply:
done